src/main.rs

```diff
@@ -1,13 +1,109 @@
-use transform_output::cargo_test::TestEvent;
-use transform_output::convert;
+use std::path::Path;
+use std::process::Command;
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let stdin = std::io::stdin();
+use anyhow::{Context, Result};
+use clap::Parser;
+use regex::Regex;
+use rust_test_runner::cargo_test::TestEvent;
+use rust_test_runner::cli::CliArgs;
+use rust_test_runner::convert;
+use serde_json as json;
 
-    let out = convert(serde_json::Deserializer::from_reader(stdin.lock()).into_iter::<TestEvent>());
+fn main() -> Result<()> {
+    let cli_args = CliArgs::parse();
+    let output_dir = cli_args
+        .output_dir
+        .canonicalize()
+        .context("failed to canonicalize output dir")?;
 
-    let stdout = std::io::stdout();
-    serde_json::to_writer_pretty(stdout.lock(), &out)?;
+    std::env::set_current_dir(&cli_args.input_dir)
+        .context("failed to 'cd' into the input directory")?;
+
+    let mut results_out = String::new();
```

Comment:
One thing I did change compared to the logic in `run.sh` is that the file `results.out` is not written to anymore. As far as I know, it is not part of the test runner interface. Internally, it is only used to store temporary data. So I left it as a mutable variable instead of writing the content to the filesystem.

Reply:
That's perfectly fine as we do indeed not use that file.